Vertical toolbar example

SomeHats opened the pull request:
Vertical toolbar! Use it like this: 

```tsx
<Tldraw components={{ Toolbar: () => <DefaultToolbar orientation="vertical" /> }} />
```

Right now, I've only supported a vertical toolbar on the left of the screen because to put it elsewhere we'd need a better way of moving other parts of the UI around. This also supports groups in the toolbar, and re-works the overflowing toolbar  component to work both horizontally and vertically.


### Change type

- [x] `api`

### Release notes

### API Changes


- You can now position the toolbar vertically down the right of the screen by setting it's orientation to `vertical`. You can also now use `<TldrawUiMenuGroup />` to group together toolbar items.

MitjaBezensek commented on packages/tldraw/src/lib/ui/components/primitives/layout.tsx:
```diff
@@ -1,10 +1,55 @@
+	tooltipSide,
+}: TldrawUiOrientationProviderProps) {
+	const prevContext = useTldrawUiOrientation()
+	const tooltipSideToUse =
```
Not sure I understand why we need to do this?

SomeHats replied:
I've added a comment here trying to explain this.

Essentially, the default tooltip side is almost always fine - but when you do change it, you then want it to cascade down until you explicitly change it again. However, if you then switch orientation your tooltip side will be wrong - it'll be covering items in your layout. So in that case, we reset it to a default.

MitjaBezensek reviewed (commented):
Nice 👏 Quite a few people wanted to have this, you're gonna make them happy 😄

Noticed a few possible UI improvements.

* Looks like there's a bit of additional space on the left side of `More` button when using groups:
<img width="372" height="256" alt="image" src="https://github.com/user-attachments/assets/589f4532-9d22-4df8-b290-c4d8fcb1dd50" />

* We overlap with the minimap in some cases. Maybe we should also have some calculation to start hiding items sooner if this happens?
<img width="478" height="514" alt="image" src="https://github.com/user-attachments/assets/9534ad01-b0b3-4f71-99c3-09dbf805f3d8" />

* Also noticed that the quick actions gets covered up by the style panel on mobile, but I think that's fine since you have to intentionally open the style panel and probably don't want to use the quick actions at the same time.
![IMG_254E86C7E7A5-1](https://github.com/user-attachments/assets/28b203ca-1790-49c5-98e8-c5f19ddca5e7)

SomeHats commented:
> Looks like there's a bit of additional space on the left side of More button when using groups:

fixed! 

> We overlap with the minimap in some cases. Maybe we should also have some calculation to start hiding items sooner if this happens?

Also fixed :)

> Also noticed that the quick actions gets covered up by the style panel on mobile, but I think that's fine since you have to intentionally open the style panel and probably don't want to use the quick actions at the same time.

yup, im gonna leave as-is for now
